earthly/docs/Earthfile

```diff
@@ -0,0 +1,175 @@
+VERSION 0.7
+FROM python:3.11-bullseye
+
+mkdocs-material:
+    # Derived from official mkdocs-material docker container.
+    # https://github.com/squidfunk/mkdocs-material/blob/master/Dockerfile
+
+    # What version is this
+    ARG tag=latest
+    ARG registry
+
+    # Installation tmp directory
+    WORKDIR /poetry
+
+    # Environment Variables
+    ENV POETRY_HOME=/poetry
+    ENV PATH=$POETRY_HOME/bin:$PATH
+    ENV PYTHONDONTWRITEBYTECODE=1
+
+    # Install extra debian packages we will need to support plugins.
+    RUN apt-get update && \
+        apt-get install -y --no-install-recommends \
+        graphviz
+
+    # Install Poetry to manage python.
+    RUN curl -sSL https://install.python-poetry.org | python3 -
+    RUN poetry config installer.max-workers 10
+
+    # Copy our dependencies.
+    COPY pyproject.toml .
+    COPY poetry.lock .
+    COPY start.sh .
+
+    # Install it all with poetry
+    RUN poetry install --only main --no-root
+
+    # Trust directory, required for git >= 2.35.2
+    RUN git config --global --add safe.directory /docs &&\
+        git config --global --add safe.directory /site
+
+    # Carry the Kroki Docker Compose with this image.
+    WORKDIR /kroki
+    COPY docker-compose.kroki.yml ./docker-compose.yml
+
+    # Set working directory
+    WORKDIR /docs
+
+    # Expose MkDocs development server port
+    EXPOSE 8000
+
+    # Start development server by default
+    ENTRYPOINT ["/poetry/start.sh"]
+    CMD ["serve", "--dev-addr=0.0.0.0:8000"]
+
+    # Finish building the container, and publish to the registry
+    IF [ "$registry" = "" ]
+        ARG registry_final=$registry
+    ELSE
+        ARG registry_final=${registry}/
+    END
+
+    SAVE ARTIFACT /kroki/docker-compose.yml /kroki/docker-compose.yml
+
+    SAVE IMAGE --push ${registry_final}mkdocs-material:${tag}
+
+MKDOCS_BUILD:
+    COMMAND
+    ARG src
+    ARG dest
+
+    # Ensure the mkdocs tools are built and up-to-date.
+    BUILD +mkdocs-material
+
+    FROM earthly/dind:alpine
+    WORKDIR /build
+    COPY +mkdocs-material/kroki/docker-compose.yml .
+
+    # Copy the src
+    COPY --dir $src /docs
+
+    # Docs need Kroki to build diagrams, so make sure its running.
+    WITH DOCKER --compose docker-compose.yml --load mkdocs=+mkdocs-material
+        RUN docker run --rm -v /docs:/docs --network=default_mkdocs mkdocs build
+    END
+
+    SAVE ARTIFACT /docs/site AS LOCAL $dest
+
+MKDOCS_SERVE:
+    COMMAND
+    ARG src
+    ARG port
+
+    # Ensure the mkdocs tools are built and up-to-date.
+    BUILD +mkdocs-material
+
+    FROM earthly/dind:alpine
+    WORKDIR /build
+    COPY +mkdocs-material/kroki/docker-compose.yml .
+
+    # Copy the src
+    COPY --dir $src /docs
+
+    # Docs need Kroki to build diagrams, so make sure its running.
+    WITH DOCKER --compose docker-compose.yml --load mkdocs=+mkdocs-material
```

Comment:
As discussed today, try:

```
LOCALLY

RUN docker compose -f docker-compose.kroki.yml
```

IIRC, your `BUILD +mkdocs-material` will have already saved the resulting image locally.

Reply:
I have reworked all this implementation to do it a different way and not try and run doc serving inside earthly.